src/proto/callbacks.proto

```diff
@@ -0,0 +1,347 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+syntax = "proto3";
+
+package lbann_data;
+
+message Callback {
```

Comment:
I would prefer if the "derived class" messages were internal to the base `Callback` message. See `Transform`:
https://github.com/LLNL/lbann/blob/96c3dbc1dfb5f20e218ca57a3c315a43a9713205/src/proto/lbann.proto#L89

Reply:
I thought about that. It's a larger change than simply moving a closing brace. In particular, I'm not sure how it fits with the python front-end. I was thinking of doing this in a separate PR. Thoughts?

Reply:
Hm, it's a good question what changes would be required. it looks like we aren't using the transforms in the Python frontend, so this approach hasn't been tested yet.

Reply:
It wasn't too much work on the C++ side after all. Keeping the names the same introduces some redundancy at the C++ level but allows the Python front-end to emerge unscathed from this. In the future, we should break everything and change the names to eliminate some of the redundancy.